`forge iOS lru-dict` fails with errors in `subprocess` call

### Describe the bug

Building after Issue #38 fix.

Running `forge iOS lru-dict` causes the following error.
```
lru.c:629:17: error: incompatible function pointer types initializing 'PyCFunction' (aka 'struct _object *(*)(struct _object *, struct _object *)') with an expression of type 'PyCFunctionWithKeywords' (aka 'struct _object *(*)(struct _object *, struct _object *, struct _object *)') [-Wincompatible-function-pointer-types]
{"popitem", (PyCFunctionWithKeywords)LRU_popitem, METH_VARARGS | METH_KEYWORDS,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
error: command '/Users/noelmullankuzhy/Tools/Python-Apple-support/support/3.11/iOS/bin/arm64-apple-ios12.0-clang' failed with exit code 1

ERROR Backend subprocess exited when trying to invoke build_wheel
********************************************************************************
Failed build: lru-dict 1.1.8 for iphoneos 12.0 on arm64
********************************************************************************
Traceback (most recent call last):
  File "/Users/noelmullankuzhy/Tools/mobile-forge/src/forge/build.py", line 268, in build
    self._build()
  File "/Users/noelmullankuzhy/Tools/mobile-forge/src/forge/build.py", line 511, in _build
    self.cross_venv.run(
  File "/Users/noelmullankuzhy/Tools/mobile-forge/src/forge/cross.py", line 356, in run
    return subprocess.run(logfile, *args, **self.cross_kwargs(kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/noelmullankuzhy/Tools/mobile-forge/src/forge/subprocess.py", line 49, in run
    raise stdlib_subprocess.CalledProcessError(return_code, args)
subprocess.CalledProcessError: Command '(['python', '-m', 'build', '--no-isolation', '--wheel', '--outdir', '/Users/noelmullankuzhy/Tools/mobile-forge/dist'],)' returned non-zero exit status 1.
```

### Steps to reproduce

1. clone repository and run `source ./setup-iOS.sh 3.11`
2. After environment is set up run `forge iOS lru-dict`


### Expected behavior

Lou-dict wheels are generated with out issues

### Screenshots

_No response_

### Environment

- Operating System: Mac OS Sonoma Version 14.3.1 (23D60)
- Python version: 3.11.6 (From Python-Apple-Support)
- Software versions:
  - Python Apple Support: branch 3.11


### Logs

Terminal Output
[Terminal Output forge iOS lri-dict Fail.txt](https://github.com/beeware/mobile-forge/files/14596286/Terminal.Output.forge.iOS.lri-dict.Fail.txt)

Error Logs
[lru-dict-1.1.8-cp311-ios_12_0_iphoneos_arm64.log](https://github.com/beeware/mobile-forge/files/14596279/lru-dict-1.1.8-cp311-ios_12_0_iphoneos_arm64.log)
[lru-dict-1.1.8-cp311-ios_12_0_iphonesimulator_arm64.log](https://github.com/beeware/mobile-forge/files/14596280/lru-dict-1.1.8-cp311-ios_12_0_iphonesimulator_arm64.log)
[lru-dict-1.1.8-cp311-ios_12_0_iphonesimulator_x86_64.log](https://github.com/beeware/mobile-forge/files/14596281/lru-dict-1.1.8-cp311-ios_12_0_iphonesimulator_x86_64.log)


### Additional context

_No response_

This is a weird one - I see the same problem, but I only get a warning, not a full error:

```
creating build/temp.ios-12.0-iphoneos-arm64-cpython-311
arm64-apple-ios12.0-clang -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -mios-version-min=12.0 -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -mios-version-min=12.0 -I/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.2.sdk/usr/include -I/Users/rkm/beeware/mobile-forge/build/cp311/lru-dict/1.1.8/venv3.11-ios_12_0_iphoneos_arm64/venv3.11-ios_12_0_iphoneos_arm64/include -I/Users/rkm/beeware/support/Python-Apple-support/install/iOS/iphoneos.arm64/python-3.11.6/include/python3.11 -c lru.c -o build/temp.ios-12.0-iphoneos-arm64-cpython-311/lru.o
lru.c:629:17: warning: incompatible function pointer types initializing 'PyCFunction' (aka 'struct _object *(*)(struct _object *, struct _object *)') with an expression of type 'PyCFunctionWithKeywords' (aka 'struct _object *(*)(struct _object *, struct _object *, struct _object *)') [-Wincompatible-function-pointer-types]
{"popitem", (PyCFunctionWithKeywords)LRU_popitem, METH_VARARGS | METH_KEYWORDS,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

So - the result you're seeing is an expected problem, but manifesting harder than it should. Ordinarily, the warning should only be promoted to an error if you're passing in `-WX` or similar. Is there any possibility that you've got an environment variable that might be doing this?

The other possibility is that there's been a change in the iPhone SDK between 17.2 and 17.4... but that seems less likely.

One way to test this out - can you build a *macOS* wheel with your current environment? Can you unpack the lru-dict tarball somewhere locally and run `python -m build --wheel` from the root of the checkout?

--- a/recipes/lru-dict/meta.yaml
+++ b/recipes/lru-dict/meta.yaml
@@ -1,3 +1,3 @@
 package:
   name: lru-dict
-  version: 1.1.8
+  version: 1.3.0

